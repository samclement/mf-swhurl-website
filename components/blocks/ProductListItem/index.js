import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'
import LazyLoad from 'react-lazy-load'

const StyledProduct = styled.div`
  width: 150px;
  min-height: 220px;
  padding: 10px;
  text-align: center;
  margin: 0;
  @media only screen and (min-width: 380px) and (max-width: 640px) {
    margin: 0 12px;
  }
  @media only screen and (min-width: 641px) and (max-width: 780px) {
    padding: 10px 5px;
    margin: 0;
  }
`

export default ({ item }) => {
  const imageUrl = `//assetsprx.matchesfashion.com/img/product/${
    item.code
  }_1_small.jpg`
  const url = `/pdp?code=${item.code}`
  return (
    <StyledProduct>
      <Link href={url} as={item.url}>
        <a aria-label={item.name}>
          <LazyLoad offsetVertical={500}>
            <img src={imageUrl} alt={item.name} width="100%" />
          </LazyLoad>
        </a>
      </Link>
      {item.name}
    </StyledProduct>
  )
}
