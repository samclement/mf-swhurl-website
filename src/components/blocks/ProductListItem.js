import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

const StyledProduct = styled.div`
  width: 150px;
  min-height: 220px;
  padding: 10px;
  text-align: center;
  margin: 0;
  @media only screen and (min-width: 380px) and (max-width: 640px) {
    margin: 0 12px;
  }
  @media only screen and (min-width: 641px) and (max-width: 780px) {
    padding: 10px 5px;
    margin: 0;
  }
`

const StyledPrice = styled.p`
  border-radius: 3px;
  border: 1px solid #ccc;
`

const StyledLink = styled.a`
  color: #000;
  &:active {
    color: #666;
  }
`

export default ({ item }) => {
  const imageUrl = `//assetsprx.matchesfashion.com/img/product/${item.code}_1_small.jpg`
  const url = `/pdp?code=${item.code}`
  return (
    <StyledProduct>
      <Link href={url} as={item.url}>
        <StyledLink aria-label={item.name}>
          <img src={imageUrl} alt={item.name} width="100%" />
          {item.name}
          <StyledPrice>
            &pound;{new Intl.NumberFormat().format(item.price)}
          </StyledPrice>
        </StyledLink>
      </Link>
    </StyledProduct>
  )
}
