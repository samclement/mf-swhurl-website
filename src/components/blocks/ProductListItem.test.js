/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import ProductListItem from './ProductListItem.js'

describe('With Snapshot Testing', () => {
  it('Product list item', () => {
    const item = { code: 1, name: 'test', price: 10 }
    const { asFragment } = render(<ProductListItem item={item} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Product list item with sale price', () => {
    const item = { code: 1, name: 'test', price: 10, previousPrice: 20 }
    const { asFragment } = render(<ProductListItem item={item} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
