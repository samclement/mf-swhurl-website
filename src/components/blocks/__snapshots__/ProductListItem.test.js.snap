// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Product list item 1`] = `
<DocumentFragment>
  .c0 {
  width: 150px;
  min-height: 220px;
  padding: 10px;
  text-align: center;
  margin: 0;
}

.c2 {
  border-radius: 3px;
  border: 1px solid #ccc;
}

.c1 {
  color: #000;
}

.c1:active {
  color: #666;
}

@media only screen and (min-width:380px) and (max-width:640px) {
  .c0 {
    margin: 0 12px;
  }
}

@media only screen and (min-width:641px) and (max-width:780px) {
  .c0 {
    padding: 10px 5px;
    margin: 0;
  }
}

<div
    class="c0"
  >
    <a
      aria-label="test"
      class="c1"
    >
      <img
        alt="test"
        src="//assetsprx.matchesfashion.com/img/product/1_1_small.jpg"
        width="100%"
      />
      test
      <p
        class="c2"
      >
        Â£10
      </p>
    </a>
  </div>
</DocumentFragment>
`;
